web-ui/pages/index.vue

```diff
@@ -44,11 +59,15 @@ export default {
       const doc = iframe.contentDocument;
 
       const logger = `
-        var console = {
+        let console = {
           log:(output) => {
-            parent.log(output);
+            parent.log(output, 'norm');
           }
         };
+
+        window.onerror = function(error, url, line) {
+          parent.log('Javascript Error : ' + error+ ' on line ' + (line - 10), 'error');
```

Comment:
Prefer template literals over string concatination:

```js
`Javascript Error : ${error} on line ${line - 10} error`
```
Also, why `line - 10`? where does this magic number come from? What happens if that changes? Is there a way co calculate the value instead?

Reply:
> the - 10 comes from the height of the logger code.
And if we ever need to change the text it you have to remember to update this magic number. It is very easy to forget or for someone else to not know it even matters.

It is best to use calculated values where you can - though in this case it is a bit awkward given it is self self-referential. One way would be to use a placeholder and `string.replace()`:

```js
const logger = `
...
parent.log('.... LINE_OFFSET ...')
...
`;
const line_count = logger.split(/\r\n|\r|\n/).length;
...
unescape("%3Cscript%3E" + logger.replace("LINE_OFFSET", line_count) + this.code + "%3C/script%3E"),
```

Reply:
Hmm, I wonder if we use two different script tags if the line count would then need to be off set at all? Might be worth a try:
```js
doc.open();
doc.write(unescape(
  `%3Cscript%3E
  ${logger}
  ${console.log(new Date(Date.now()).toLocaleTimeString())}
  %3C/script%3E`
));
doc.write(unescape(
  `%3Cscript%3E
  ${this.code}
  %3C/script%3E`
));
doc.write(unescape(
  `%3Cscript%3E
  ${console.log('\\n')}
  %3C/script%3E`
));
doc.close();
```